test/spec/spectest.cpp

```diff
@@ -562,116 +571,118 @@ void SpecTest::run(std::string_view Proposal, std::string_view UnitName) {
+      const simdjson::dom::array Expected = Cmd["expected"];
+      std::string_view ActType = Action["type"];
+      if (ActType == "invoke"sv) {
+        Invoke(Action, Expected, LineNumber);
         return;
-      }
-      case CommandID::AssertInvalid: {
-        const auto Filename = (TestsuiteRoot / Proposal / UnitName /
-                               Cmd["filename"s].Get<std::string>())
-                                  .u8string();
-        const auto &Text = Cmd["text"s].Get<std::string>();
-        TrapValidate(Filename, Text);
+      } else if (ActType == "get"sv) {
+        Get(Action, Expected, LineNumber);
         return;
       }
-      case CommandID::AssertUnlinkable:
-      case CommandID::AssertUninstantiable: {
-        const auto Filename = (TestsuiteRoot / Proposal / UnitName /
-                               Cmd["filename"s].Get<std::string>())
-                                  .u8string();
-        const auto &Text = Cmd["text"s].Get<std::string>();
-        TrapInstantiate(Filename, Text);
+      EXPECT_TRUE(false);
+      return;
+    }
+    case CommandID::AssertTrap: {
+      const simdjson::dom::object Action = Cmd["action"].get_object();
+      const std::string_view Text = Cmd["text"];
+      const uint64_t LineNumber = Cmd["line"];
+      TrapInvoke(Action, std::string(Text), LineNumber);
+      return;
+    }
+    case CommandID::AssertExhaustion: {
+      // TODO: Add stack overflow mechanism.
+      return;
+    }
+    case CommandID::AssertMalformed: {
+      std::string_view ModType = Cmd["module_type"];
+      std::string Binary = "binary";
+      if (Binary.compare(std::string(ModType))) {
+        // TODO: Wat is not supported in WasmEdge yet.
         return;
       }
-      default:;
-      }
+      std::string_view Name = Cmd["filename"];
+      const auto Filename =
+          (TestsuiteRoot / Proposal / UnitName / Name).u8string();
+      Name = Cmd["text"];
+      TrapLoad(Filename, std::string(Name));
+      return;
+    }
+    case CommandID::AssertInvalid: {
+      std::string_view Name = Cmd["filename"];
+      const auto Filename =
+          (TestsuiteRoot / Proposal / UnitName / Name).u8string();
+      Name = Cmd["text"];
+      TrapValidate(Filename, std::string(Name));
+      return;
+    }
+    case CommandID::AssertUnlinkable:
+    case CommandID::AssertUninstantiable: {
+      std::string_view Name = Cmd["filename"];
+      const auto Filename =
+          (TestsuiteRoot / Proposal / UnitName / Name).u8string();
+      Name = Cmd["text"];
+      TrapInstantiate(Filename, std::string(Name));
+      return;
+    }
+    default:;
     }
     // Unknown command.
     EXPECT_TRUE(false);
   };
 
   // Get command list.
-  if (auto Commands = Doc.FindMember("commands"s);
-      Commands != Doc.MemberEnd()) {
-    rapidjson::Value CmdArray;
-    CmdArray.CopyFrom(Commands->value, Allocator);
+  simdjson::dom::array CmdArray = Doc["commands"];
```

Comment:
What if the `commands` is not in `Doc`? We should have a check for this.

Reply:
Ohh, didn't consider this. Will add a check for the same.

Reply:
I have added the check for commands list with [e7162ac](https://github.com/WasmEdge/WasmEdge/pull/2659/commits/e7162acd4ff9812fa00a628c6d35802cb09d6ea6)